pupil_src/shared_modules/task_lib/manager.py

```diff
@@ -0,0 +1,126 @@
+    Parts of your plugin don't need to manage tasks themselves but get
+    called when something interesting happens.
+
+    How to use:
+    1) Make your plugin an Observable.
+    2) Create a PluginTaskManager() in your plugin.
+    3) Give the manager e.g. to controllers and other parts of your plugin that need
+       to start tasks.
+    4) Call create_background_task() to add tasks, or add you own task that inherits
+       from TaskInterface via add_task().
+    """
+
+    def __init__(self, plugin):
+        self._scheduled_tasks = []
+        self._running_tasks = []
+        plugin.add_observer("recent_events", self.on_recent_events)
+        plugin.add_observer("cleanup", self.on_cleanup)
+
+    def __del__(self):
+        self.on_cleanup()
+
+    def create_background_task(
+        self,
+        name,
+        routine_or_generator_function,
+        pass_shared_memory=False,
+        args=None,
+        kwargs=None,
+        patches=(IPCLoggingPatch(),),
+    ):
+        """
+        Creates a managed background task.
+
+        The task is started automatically during the next recent_events.
+        If you want to start it immediately, you can call start() on the returned task.
+
+        Right after you created the task, you can call add_observer on the returned
+        task to get notified when your task completes, yields results etc.
+        See task_lib.interface.py for a list of all events you can add observers to.
+
+        Args:
+            name (String): Name of the process for the task, only for identification
+                purposes. Multiple processes might have the same name.
+            routine_or_generator_function (Callable): Your task. Supports generator
+                functions (= methods and functions that yield results) and routines
+                (= methods and functions without yield).
+            args: Arguments for the task.
+            kwargs: Keyword arguments for the task.
+            pass_shared_memory: If True, a shared memory object will be passed to the
+                task (see task_lib.background.shared_memory.py). For this to work,
+                your task needs to accept a keyword argument `shared_memory`.
+            patches (collection of Patch): Patches that will be applied to your
+                background process right after start. Use them if you need to fix
+                something in the environment of the new process (see
+                task_lib.background.patches.py).
+                Per default, the IPC logging is patched.
+
+        Returns:
+            A new task with base class TaskInterface.
+
+        """
+        task = task_lib.background.create(
+            name,
+            routine_or_generator_function,
+            pass_shared_memory,
+            args,
+            kwargs,
+            patches,
+        )
+        self._scheduled_tasks.append(task)
+        return task
+
+    def add_task(self, task):
```

Comment:
Well, you can already start it before you call this. Yes, `add` is a little ambiguous, but I hoped it would be not too surprising that a manager is starting a task you added if you didn't. Thinking about this, I should probably rename `scheduled_tasks` to `fresh_tasks` or what do you think?

Reply:
`waiting_tasks`, `recently_added_tasks`, `recently_scheduled_tasks`?

Reply:
Thanks, good suggestions!